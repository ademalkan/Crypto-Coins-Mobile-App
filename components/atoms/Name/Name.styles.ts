import { StyleSheet } from "react-native";

export default StyleSheet.create({
  name: {
    fontSize: 16,
    fontWeight: 'bold',
    marginRight: 5
  },
})