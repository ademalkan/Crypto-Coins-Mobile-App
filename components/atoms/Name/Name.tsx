import React from 'react';
import { Text } from 'react-native';
import styles from "./Name.styles";

type NameProps = {
  name: string;
};

const Name: React.FC<NameProps> = ({ name }) => {
  return (
    <Text style={styles.name}>{name}</Text>
  );
};

export default Name;
