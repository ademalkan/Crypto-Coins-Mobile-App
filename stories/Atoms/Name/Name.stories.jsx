import React from "react";
import { View } from "react-native";
import Name from "../../../components/atoms/Name/Name";

export default {
  title: "Atoms/Name",
  component: Name,
  args: {
    name: "Name Example",
  },
  decorators: [
    (Story) => (
      <View style={{ alignItems: "center", justifyContent: "center", flex: 1 }}>
        <Story />
      </View>
    ),
  ],
};

export const Default = ({ ...args }) => <Name {...args} />;
