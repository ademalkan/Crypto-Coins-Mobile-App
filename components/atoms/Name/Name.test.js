import React from "react";
import renderer from "react-test-renderer";
import Name from "./Name";

describe("Name", () => {
  it("should render the name prop correctly", () => {
    const name = "Adem ALKAN";

    const component = renderer.create(<Name name={name} />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
