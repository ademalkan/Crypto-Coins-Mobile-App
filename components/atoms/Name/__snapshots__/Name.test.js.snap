// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Name should render the name prop correctly 1`] = `
<Text
  style={
    {
      "fontSize": 16,
      "fontWeight": "bold",
      "marginRight": 5,
    }
  }
>
  Adem ALKAN
</Text>
`;
